playbooks/roles/mattermost/templates/config.json.j2

```diff
@@ -90,46 +98,59 @@
         "RestrictPublicChannelDeletion": "all",
         "RestrictPrivateChannelDeletion": "all",
         "RestrictPrivateChannelManageMembers": "all",
-        "EnableXToLeaveChannelsFromLHS": false,
         "UserStatusAwayTimeout": 300,
         "MaxChannelsPerTeam": 2000,
-        "MaxNotificationsPerChannel": 1000000,
-        "EnableConfirmNotificationsToChannel": true,
-        "TeammateNameDisplay": "username",
-        "ExperimentalEnableAutomaticReplies": false,
-        "ExperimentalHideTownSquareinLHS": false,
-        "ExperimentalTownSquareIsReadOnly": false,
-        "ExperimentalPrimaryTeam": ""
-    },
-    "ClientRequirements": {
+        "MaxNotificationsPerChannel": {{ MATTERMOST_MAX_CHANNEL_NOTIFICATIONS }},
+        "TeammateNameDisplay": "{{ MATTERMOST_TEAMMATE_NAME }}",
+        {% if {{ MATTERMOST_VERSION }} != '4.0.0' %}
+            "EnableXToLeaveChannelsFromLHS": false,
+            "EnableConfirmNotificationsToChannel": true,
+            "ExperimentalEnableAutomaticReplies": false,
+            "ExperimentalHideTownSquareinLHS": false,
+            "ExperimentalTownSquareIsReadOnly": false,
+            "ExperimentalPrimaryTeam": ""
+    },
+    {% if {{ MATTERMOST_VERSION }} != '4.0.0' %}
+        "ClientRequirements": {	
         "AndroidLatestVersion": "",
         "AndroidMinVersion": "",
         "DesktopLatestVersion": "",
         "DesktopMinVersion": "",
         "IosLatestVersion": "",
         "IosMinVersion": ""
-    },
+        },
     "SqlSettings": {
         "DriverName": "postgres",
         "DataSource": "{{ MATTERMOST_POSTGRES_URL }}",
         "DataSourceReplicas": [],
         "DataSourceSearchReplicas": [],
-        "MaxIdleConns": 20,
-        "ConnMaxLifetimeMilliseconds": 3600000,
-        "MaxOpenConns": 300,
+        {% if {{ MATTERMOST_VERSION }} == '4.0.0' %}
+            "MaxIdleConns": 10,
+            "MaxOpenConns": 10,
+        {% else %}
+            "MaxIdleConns": 20,
+            "ConnMaxLifetimeMilliseconds": 3600000,
+            "MaxOpenConns": 300,
         "Trace": false,
         "AtRestEncryptKey": "{{ MATTERMOST_POSTGRES_ENCRYPT_KEY }}",
         "QueryTimeout": 30
     },
     "LogSettings": {
-        "EnableConsole": false,
-        "ConsoleLevel": "INFO",
-        "ConsoleJson": true,
+        {% if {{ MATTERMOST_VERSION }} == '4.0.0' %}
+            "EnableConsole": true,
+            "ConsoleLevel": "DEBUG",
+        {% else %}
+            "EnableConsole": false,
+            "ConsoleLevel": "INFO",
+            "ConsoleJson": true,
         "EnableFile": true,
         "FileLevel": "INFO",
-        "FileJson": true,
+        {% if {{ MATTERMOST_VERSION }} == '4.0.0' %}
+            "FileFormat": "",
+        {% else %}
+            "FileJson": true,
         "FileLocation": "{{ MATTERMOST_LOG_DIR }}",
-        "EnableWebhookDebugging": false,
+        "EnableWebhookDebugging": true,
```

Comment:
This line isn't protected by a variable.

Reply:
Modified it to use a variable.